### DISCUSSION TITLE
What is the best way to get angular state feedback?

### DISCUSSION THREAD

**Discussion by Judy021** [USC-ACTLab/crazyswarm#762] [Q&A]
I'm using cf.cmdFullState to follow a path computed using LQR. I need full state feedback to compute the optimal control to follow the path. Th problem is there is no way to get attitude feedback (roll, pitch, yaw angles and rate).  I tried using cf.state() but I get the following error:
    current_state = cf.state()
TypeError: 'traj_eval' object is not callable

I can obtain linear position and velocity using cf.position() and cf.velocity() but how do I get angular state information?

**Comment by knmcguire** [comment]
Not sure if there is an API for that... but you could use the logging <-> topics frameworks perhaps? That is something you can set in the launch file and you can then observe these logging variables: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#stateestimate
